File: backend-simple.js
const express = require('express');
const cors = require('cors');
const app = express();
const port = 8080;

// Middleware
app.use(cors());
app.use(express.json());

// Stockage temporaire des utilisateurs (en mémoire)
let users = [];

// Endpoint d'inscription
app.post('/api/users/register', (req, res) => {
    const { username, email, password, poids, taille, age } = req.body;

    // Validation basique
    if (!username || !email || !password || !poids || !taille || !age) {
        return res.status(400).json({ error: 'Tous les champs sont requis' });
    }

    if (password.length < 6) {
        return res.status(400).json({ error: 'Le mot de passe doit contenir au moins 6 caractères' });
    }

    // Vérifier si l'utilisateur existe déjà
    const existingUser = users.find(user => user.email === email);
    if (existingUser) {
        return res.status(400).json({ error: 'Cet email est déjà utilisé' });
    }

    // Créer l'utilisateur
    const newUser = {
        id: users.length + 1,
        username,
        email,
        password, // En production, il faudrait hasher le mot de passe
        poids: parseFloat(poids),
        taille: parseInt(taille),
        age: parseInt(age),
        createdAt: new Date()
    };

    users.push(newUser);

    console.log('Nouvel utilisateur inscrit:', newUser);

    res.status(200).json({
        id: newUser.id,
        username: newUser.username,
        email: newUser.email,
        poids: newUser.poids,
        taille: newUser.taille,
        age: newUser.age
    });
});

// Endpoint de connexion
app.post('/api/users/login', (req, res) => {
    const { email, password } = req.body;

    if (!email || !password) {
        return res.status(400).json({ error: 'Email et mot de passe requis' });
    }

    // Trouver l'utilisateur
    const user = users.find(u => u.email === email && u.password === password);

    if (!user) {
        return res.status(401).json({ error: 'Email ou mot de passe incorrect' });
    }

    console.log('Utilisateur connecté:', user.email);

    res.status(200).json({
        id: user.id,
        username: user.username,
        email: user.email,
        poids: user.poids,
        taille: user.taille,
        age: user.age
    });
});

// Endpoint pour obtenir tous les utilisateurs (pour debug)
app.get('/api/users', (req, res) => {
    res.json(users);
});

// Démarrer le serveur
app.listen(port, () => {
    console.log(`🚀 Serveur backend démarré sur http://localhost:${port}`);
    console.log(`📝 Endpoint inscription: POST http://localhost:${port}/api/users/register`);
    console.log(`🔐 Endpoint connexion: POST http://localhost:${port}/api/users/login`);
    console.log(`👥 Liste utilisateurs: GET http://localhost:${port}/api/users`);
});

File: backend_simple.py
from flask import Flask, request, jsonify
from flask_cors import CORS
import json
from datetime import datetime

app = Flask(__name__)
CORS(app)  # Permettre les requêtes CORS

# Stockage temporaire des utilisateurs (en mémoire)
users = []

@app.route('/api/users/register', methods=['POST'])
def register():
    data = request.get_json()

    # Validation des champs requis
    required_fields = ['username', 'email', 'password', 'poids', 'taille', 'age']
    for field in required_fields:
        if field not in data:
            return jsonify({'error': f'Le champ {field} est requis'}), 400

    username = data['username']
    email = data['email']
    password = data['password']
    poids = data['poids']
    taille = data['taille']
    age = data['age']

    # Validation du mot de passe
    if len(password) < 6:
        return jsonify({'error': 'Le mot de passe doit contenir au moins 6 caractères'}), 400

    # Vérifier si l'utilisateur existe déjà
    for user in users:
        if user['email'] == email:
            return jsonify({'error': 'Cet email est déjà utilisé'}), 400

    # Créer l'utilisateur
    new_user = {
        'id': len(users) + 1,
        'username': username,
        'email': email,
        'password': password,  # En production, il faudrait hasher le mot de passe
        'poids': float(poids),
        'taille': int(taille),
        'age': int(age),
        'created_at': datetime.now().isoformat()
    }

    users.append(new_user)

    print(f'✅ Nouvel utilisateur inscrit: {new_user["email"]}')

    # Retourner l'utilisateur sans le mot de passe
    response_user = {k: v for k, v in new_user.items() if k != 'password'}
    return jsonify(response_user), 200

@app.route('/api/users/login', methods=['POST'])
def login():
    data = request.get_json()

    if 'email' not in data or 'password' not in data:
        return jsonify({'error': 'Email et mot de passe requis'}), 400

    email = data['email']
    password = data['password']

    # Trouver l'utilisateur
    for user in users:
        if user['email'] == email and user['password'] == password:
            print(f'✅ Utilisateur connecté: {email}')
            # Retourner l'utilisateur sans le mot de passe
            response_user = {k: v for k, v in user.items() if k != 'password'}
            return jsonify(response_user), 200

    return jsonify({'error': 'Email ou mot de passe incorrect'}), 401

@app.route('/api/users', methods=['GET'])
def get_users():
    # Retourner tous les utilisateurs sans les mots de passe
    safe_users = [{k: v for k, v in user.items() if k != 'password'} for user in users]
    return jsonify(safe_users), 200

if __name__ == '__main__':
    print("🚀 Serveur backend Python démarré sur http://localhost:5000")
    print("📝 Endpoint inscription: POST /api/users/register")
    print("🔐 Endpoint connexion: POST /api/users/login")
    print("👥 Liste utilisateurs: GET /api/users")
    app.run(debug=True, port=5000, host='0.0.0.0')